// ngx-wig/esm2015/lib/config.js
import { InjectionToken } from '@angular/core';
export const DEFAULT_LIBRARY_BUTTONS = {
    list1: {
        label: 'UL',
        title: 'Unordered List',
        command: 'insertunorderedlist',
        styleClass: 'list-ul',
        icon: 'icon-list-ul'
    },
    list2: {
        label: 'OL',
        title: 'Ordered List',
        command: 'insertorderedlist',
        styleClass: 'list-ol',
        icon: 'icon-list-ol'
    },
    bold: {
        label: 'B',
        title: 'Bold',
        command: 'bold',
        styleClass: 'bold',
        icon: 'icon-bold'
    },
    italic: {
        label: 'I',
        title: 'Italic',
        command: 'italic',
        styleClass: 'italic',
        icon: 'icon-italic'
    },
    link: {
        label: 'Link',
        title: 'Link',
        command: 'createlink',
        styleClass: 'link',
        icon: 'icon-link'
    },
    underline: {
        label: 'U',
        title: 'Underline',
        command: 'underline',
        styleClass: 'format-underlined',
        icon: 'icon-underline'
    }
};
const ɵ0 = (ctx) => {
    ctx.editMode = !ctx.editMode;
};
export const CUSTOM_LIBRARY_BUTTONS = {
    edithtml: {
        label: 'Edit HTML',
        title: 'Edit HTML',
        command: ɵ0,
        styleClass: 'nw-button--source',
        icon: '',
    }
};
export const BUTTONS = new InjectionToken('BUTTONS');
export { ɵ0 };
