// ngx-wig/esm2015/lib/ngx-wig.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxWigComponent } from './ngx-wig.component';
import { FormsModule } from '@angular/forms';
import { DEFAULT_LIBRARY_BUTTONS, BUTTONS } from './config';
export function getWindowObject() {
    return window;
}
const ɵ0 = DEFAULT_LIBRARY_BUTTONS;
export class NgxWigModule {
    static forRoot(config) {
        return {
            ngModule: NgxWigModule,
            providers: [
                {
                    provide: BUTTONS,
                    multi: true,
                    useValue: ((!config || !(config === null || config === void 0 ? void 0 : config.buttonsConfig)) ? DEFAULT_LIBRARY_BUTTONS : config === null || config === void 0 ? void 0 : config.buttonsConfig)
                },
                { provide: 'WINDOW', useFactory: getWindowObject },
            ],
        };
    }
    static forChild() {
        return { ngModule: NgxWigModule };
    }
}
NgxWigModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    NgxWigComponent
                ],
                imports: [
                    CommonModule,
                    FormsModule,
                ],
                exports: [NgxWigComponent],
                providers: [
                    { provide: BUTTONS, multi: true, useValue: ɵ0 },
                    { provide: 'WINDOW', useFactory: getWindowObject },
                ]
            },] }
];
export { ɵ0 };
