// ngx-wig/esm2015/lib/ngx-wig-toolbar.service.js
import { Injectable, Inject } from '@angular/core';
import { BUTTONS } from './config';
import * as i0 from "@angular/core";
import * as i1 from "./config";
export class NgxWigToolbarService {
    constructor(buttonLibraryConfig) {
        this._defaultButtonsList = [];
        this._buttonLibrary = buttonLibraryConfig.reduce((acc, val) => (Object.assign(Object.assign({}, acc), val)), {});
        this._defaultButtonsList = Object.keys(this._buttonLibrary);
    }
    setButtons(buttons) {
        if (!Array.isArray(buttons)) {
            throw new Error('Argument "buttons" should be an array');
        }
        this._defaultButtonsList = buttons;
    }
    addStandardButton(name, title, command, styleClass, icon) {
        if (!name || !title || !command) {
            throw new Error('Arguments "name", "title" and "command" are required');
        }
        styleClass = styleClass || '';
        this._buttonLibrary[name] = { title, command, styleClass, icon };
        this._defaultButtonsList.push(name);
    }
    getToolbarButtons(buttonsList) {
        let buttons = this._defaultButtonsList;
        const toolbarButtons = [];
        if (typeof buttonsList !== 'undefined') {
            buttons = string2array(buttonsList);
        }
        buttons.forEach(buttonKey => {
            if (!buttonKey) {
                return;
            }
            if (!this._buttonLibrary[buttonKey]) {
                throw new Error(`There is no "${buttonKey}" in your library. Possible variants: ${Object.keys(this._buttonLibrary)}`);
            }
            const button = Object.assign({}, this._buttonLibrary[buttonKey]);
            // button.isActive = () => {return !!this.command && document.queryCommandState(this.command);}
            toolbarButtons.push(button);
        });
        return toolbarButtons;
    }
}
NgxWigToolbarService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgxWigToolbarService_Factory() { return new NgxWigToolbarService(i0.ɵɵinject(i1.BUTTONS)); }, token: NgxWigToolbarService, providedIn: "root" });
NgxWigToolbarService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
NgxWigToolbarService.ctorParameters = () => [
    { type: Array, decorators: [{ type: Inject, args: [BUTTONS,] }] }
];
function string2array(keysString) {
    return keysString
        .split(',')
        .map(Function.prototype.call, String.prototype.trim);
}
